picocli-codegen/src/main/java/picocli/codegen/docgen/manpage/ManPageGenerator.java

```diff
@@ -441,12 +449,14 @@ private static void writeOption(PrintWriter pw, IOptionRenderer optionRenderer,
     }
 
     private static void writePositional(PrintWriter pw, PositionalParamSpec positional, IParameterRenderer parameterRenderer, IParamLabelRenderer paramLabelRenderer) {
-        pw.println();
-        Text[][] rows = parameterRenderer.render(positional, paramLabelRenderer, COLOR_SCHEME);
-        pw.printf("%s::%n", join(", ", rows[0][1], rows[0][3]));
-        pw.printf("  %s%n", rows[0][4]);
-        for (int i = 1; i < rows.length; i++) {
-            pw.printf("+%n%s%n", rows[i][4]);
+        if (!positional.hidden()) {
```

Comment:
Should we have a similar if statement in `writeOption` : `if (!option.hidden()) ...`, 
or should we exclude hidden `PositionalParamSpec` before the `writePositional` method is called?
I would like it if we can do this in a consistent manner for options, positionals and commands.

Reply:
+1, keeping the check in `writeOption`/`writePositional` is easier, as `writePositional` is called from multiple places.

However, the check for the subcommands gets a little trickier, the subcommands need to be removed before processing to avoid the tag/"Commands" heading.

Thoughts?

Reply:
After thinking about this some more, I think your original approach to filter out hidden args first is better.

The reason is the same as for subcommands: if all options or positional parameters are hidden, we don't want to include the tag and heading in the output. (see also https://github.com/remkop/picocli/issues/1071)